io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -102,31 +102,44 @@ public boolean isCancelable() {
 	@Override
 	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 			throws IOProviderConfigurationException, IOException {
-		progress.begin("Generating Shapefile", ProgressIndicator.UNKNOWN);
+		progress.begin("Generating Shapefile/CPG", ProgressIndicator.UNKNOWN);
 		InstanceCollection instances = getInstances();
-		List<String> filesWritten;
-		try {
-			URI location = getTarget().getLocation();
-			String filePath = Paths.get(location).getParent().toString();
-
-			filesWritten = writeInstances(instances, progress, reporter, location);
-
-			if (filesWritten.size() > 1) {
-				List<URI> uris = filesWritten.stream().map(f -> {
-					File file = new File(filePath + "/" + f + ShapefileConstants.SHP_EXTENSION);
-					return file.toURI();
-				}).collect(Collectors.toList());
+		/**
```

Comment:
This looks strange - two new methods defined in another method? Does that compile for you?
Also, the removed code looks quite important to the functionality to me.

I see the there is an existing class for testing writing Shapefiles (see [here](https://github.com/halestudio/hale/blob/master/io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy)). It would be great if you could extend the test cases to verify that the CPG file is written and contains the expected content.

Reply:
This is compiling and it's working.

The second new method is called from 2 places: once when you have more than one source and once when you have just one source data/feature type. 
There was no code removed. Please check bellow and you will see that you can still find it. The for has been changes to accommodate the creation of .cpg as well. 

I will extend the test to verify the .cpg

Reply:
> This is compiling and it's working.

Well, I reviewed the version before you added the additional commit, so I could only take into account what I could see here.

It would be great if you can combine the two commits, since the first one seems to have been corrupt in some way then.